libp2p/pubsub/gossipsub.py

```diff
@@ -121,17 +116,13 @@ def add_peer(self, peer_id: ID, protocol_id: TProtocol) -> None:
         """
         logger.debug("adding peer %s with protocol %s", peer_id, protocol_id)
 
-        if protocol_id == PROTOCOL_ID:
-            self.peers_gossipsub.append(peer_id)
-        elif protocol_id == floodsub.PROTOCOL_ID:
-            self.peers_floodsub.append(peer_id)
-        else:
+        if protocol_id not in (PROTOCOL_ID, floodsub.PROTOCOL_ID):
             # We should never enter here. Becuase the `protocol_id` is registered by your pubsub
             #   instance in multistream-select, but it is not the protocol that gossipsub supports.
             #   In this case, probably we registered gossipsub to a wrong `protocol_id`
             #   in multistream-select, or wrong versions.
             raise Exception(f"Unreachable: Protocol={protocol_id} is not supported.")
```

Comment:
We might able to make this unreachable exception a ValueError now
```suggestion
             raise ValueError(f"Protocol={protocol_id} is not supported.")
```